osquery/tables/system/darwin/homebrew_packages.cpp

```diff
@@ -56,10 +56,57 @@ std::vector<std::string> getHomebrewVersionsFromInfoPlistPath(
   return results;
 }
 
+Status getHomebrewPrefix(fs::path& prefix) {
+  // The Homebrew wrapper script finds the Library directory by taking the
+  // directory that it is located in and concatenating `/../Library`:
+  //   BREW_FILE_DIRECTORY=$(chdir "${0%/*}" && pwd -P)
+  //   export HOMEBREW_BREW_FILE="$BREW_FILE_DIRECTORY/${0##*/}"
+  // Note that the `-P` flag to pwd resolves all symlinks.
+  //
+  // Next, it will use given filename to find the prefix:
+  //   HOMEBREW_PREFIX = Pathname.new(HOMEBREW_BREW_FILE).dirname.parent
+
+  if (!pathExists(kHomebrewBinary).ok()) {
+    return Status(1, "No Homebrew binary found");
+  }
+
+  // Get the actual location of the Homebrew binary.
+  // In the future, we could extend this to look at all 'brew' executables in
+  // $PATH and check all of them.
+  auto brewExecutable = fs::canonical("/usr/local/bin/brew");
+
+  // Note that the first `parent_path` call is to remove the filename, and the
+  // next to actually move up a directory.
+  prefix = brewExecutable.parent_path().parent_path();
+  return Status(0, "OK");
+}
+
+Status getHomebrewCellar(fs::path& cellarPath) {
+  fs::path path;
+
+  auto stat = getHomebrewPrefix(path);
```

Comment:
No need to break this into two methods. Just find the Cellar and yield to an output argument in a single method.